journal/202106.md

```diff
@@ -0,0 +1,467 @@
+- ~10 bug fixes
+
+Developer-facing changes include the completion of several security improvements that have been long in the making:
+
+- automated tests for [Send](https://github.com/decred/decrediton/pull/3489), [Receive](https://github.com/decred/decrediton/pull/3490), and [Export](https://github.com/decred/decrediton/pull/3493) views
+- reorganized, cleaned up and optimized [Webpack config](https://github.com/decred/decrediton/pull/3503)
+- disabled [Node Integration](https://github.com/decred/decrediton/pull/3486) in the UI code so that it can no longer access the low level Node and Electron APIs directly. This is an important security feature and a [recommended](https://www.electronjs.org/docs/tutorial/security#2-do-not-enable-nodejs-integration-for-remote-content) practice in modern Electron app development.
+- [context isolation](https://github.com/decred/decrediton/pull/3492) enabled in main window - another important [security](https://www.electronjs.org/docs/tutorial/security#3-enable-context-isolation-for-remote-content) feature further restricting the UI code
+- enabled the [webSecurity](https://github.com/decred/decrediton/pull/3500) feature to tighten up external requests (this also fixed the development mode on Windows)
+- reduced [dependencies](https://github.com/decred/decrediton/pull/3509) to rely less on third party code and improve security by reducing the attack surface of a possibly supply chain attack
+- introduced a more secure confirmation dialog and used it for confirming [VSP access](https://github.com/decred/decrediton/pull/3515) and [transaction signing](https://github.com/decred/decrediton/pull/3519)
+
+{ UI shot of some view added/updated in June }
+
+<a id="politeia" />
+
+**[Politeia](https://github.com/decred/politeia)**
+
+User-facing:
+
+- added a button to access [raw Markdown](https://github.com/decred/politeiagui/issues/2415)
+- show status of [proposal credit](https://github.com/decred/politeiagui/pull/2417) payment (gives an idea how soon the credits can be used to submit new proposals)
+- show a [banner](https://github.com/decred/politeiagui/pull/2456) when the user is viewing `proposals-archive.decred.org` to reduce confusion when clicking site logo
+- improved look of the [Flat mode](https://github.com/decred/politeiagui/pull/2434) button
+- added proposal name to email [subjects](https://github.com/decred/politeia/pull/1440)
+- ~13 bug fixes
+
+Backend and CLI:
+
+- `politeiavoter` [upgraded](https://github.com/decred/politeia/pull/1235) to the new API released with [v1.0.0](https://github.com/decred/politeia/releases/tag/v1.0.0)
+- `politeiavoter` will now [retry](https://github.com/decred/politeia/pull/1438) its votes on server errors
+- added [MySQL](https://github.com/decred/politeia/pull/1419) implementation for the user database (the backend uses MySQL because of [tlog](https://github.com/decred/politeia/issues/1392))
+  - { can also comment why we need to migrate from cockroach to mysql }
```

Comment:
> { can also comment why we need to migrate from cockroach to mysql }

The production site currently uses CockroachDB for the user database.  Getting rid of this dependency requires setting up the new MySQL implemenation then migrating the existing data from CockroachDB to MySQL.

Reply:
Thanks. I assume the intent is to simply deal with one db and not two. Changed to this text for now:

> added [MySQL](https://github.com/decred/politeia/pull/1419) implementation for the user database. The data will need to be migrated from CockroachDB to remove this dependency. The choice was limited to MySQL by [tlog](https://github.com/decred/politeia/issues/1392).

Reply:
Correct.  Ideally, everthing would be running on postgres, but tlog is the limiting factor.